downstairs/src/lib.rs

```diff
@@ -3308,45 +3312,35 @@ impl Work {
     /// Updates `self.outstanding_deps` and prints a warning message the first
     /// time a job with unmet dependencies is checked.
     fn check_ready(&mut self, ds_id: JobId, work: &mut IOop) -> bool {
-        /*
-         * Before we can make this in_progress, we have to check the dep
-         * list if there is one and make sure all dependencies are
-         * completed.
-         */
-        let dep_list = work.deps();
-
-        /*
-         * See which of our dependencies are met.
-         * XXX Make this better/faster by removing the ones that
-         * are met, so next lap we don't have to check again?  There
-         * may be some debug value to knowing what the dep list was,
-         * so consider that before making this faster.
-         */
-        let mut deps_outstanding: Vec<JobId> =
-            Vec::with_capacity(dep_list.len());
-
-        for dep in dep_list.iter() {
-            // The Downstairs currently assumes that all jobs previous
-            // to the last flush have completed, hence this early out.
-            //
-            // Currently `work.completed` is cleared out when
-            // `ActiveConnection::do_ready_work` (or `complete` in mod test) is
-            // called with a `FlushAck`, so this early out cannot be removed
-            // unless that is changed too.
-            if dep <= &self.last_flush {
-                continue;
-            }
-
-            if !self.completed.contains(dep) {
-                deps_outstanding.push(*dep);
-            }
-        }
+        // Before we can make this in_progress, we have to check the dep
```

Comment:
With the recent changes, we don't make things "in_progress" any more, so the
language here is a little strange.  Perhaps this could read:
`Before we can perform an IO, we have to check ...`

Reply:
Also updated in beea9bc1